Docs tutorial section (#898)
quickwit-inc/quickwit: docs/tutorials/tutorial-hdfs-logs.md

```yaml title="hdfslogs_index_config.yaml"
version: 0

doc_mapping:
  field_mappings:
    - name: timestamp
      type: i64
      fast: true # Fast field must be present when this is the timestamp field.
    - name: severity_text
      type: text
      tokenizer: raw # No tokeninization.
    - name: body
      type: text
      tokenizer: default
      record: position # Record position will enable phrase query on body field.
    - name: resource
      type: object
      field_mappings:
        - name: service
          type: text
          tokenizer: raw # Text field referenced as tag must have the `raw` tokenier.
  tag_fields: [resource.service]
  store_source: true

indexing_settings:
  timestamp_field: timestamp

search_settings:
  default_search_fields: [severity_text, body]
```

Now let's create the index with the `create` subcommand (assuming you are inside quickwit install directory):

Let's make use of the default configuration file provided in quickwit installation throughout this guide by setting the `QW_CONFIG` environment variable.

```bash

> export QW_CONFIG=./config/quickwit.yaml

Reviewer: Add a small comment just to say that all commands need quickwit config and it's a good practice to have it in this env variable.

```bash
curl -L https://install.quickwit.io | sh
cd quickwit-v*/
```

## Create your index

Let's create an index configured to receive these logs.

```bash
# First, download the hdfs logs config from Quickwit repository.
curl -o hdfslogs_index_config.yaml https://raw.githubusercontent.com/quickwit-inc/quickwit/main/config/tutorials/hdfs-logs/index-config.yaml
```

The index config defines four fields: `timestamp`, `severity_text`, `body`, and one object field

> for the nested values `resource.service` . It also sets the `default_search_fields`, the `tag_fields`, and the `timestamp_field`.The `timestamp_field` and `tag_fields` are used by Quickwit for [splits pruning](../overview/architecture.md) at query time to boost search speed. Check out the [index config docs](../reference/index-config.md) for more details.

Reviewer: ```suggestion
for the nested values `resource.service` . It also sets the `default_search_fields`, the `tag_fields`, and the `timestamp_field`. The `timestamp_field` and `tag_fields` are used by Quickwit for [splits pruning](../overview/architecture.md) at query time to boost search speed. Check out the [index config docs](../reference/index-config.md) for more details.
```

```yaml title="hdfslogs_index_config.yaml"
version: 0

doc_mapping:
  field_mappings:
    - name: timestamp
      type: i64
      fast: true # Fast field must be present when this is the timestamp field.
    - name: severity_text
      type: text
      tokenizer: raw # No tokeninization.
    - name: body
      type: text
      tokenizer: default
      record: position # Record position will enable phrase query on body field.
    - name: resource
      type: object
      field_mappings:
        - name: service
          type: text
          tokenizer: raw # Text field referenced as tag must have the `raw` tokenier.
  tag_fields: [resource.service]
  store_source: true

indexing_settings:
  timestamp_field: timestamp

search_settings:
  default_search_fields: [severity_text, body]
```

Now let's create the index with the `create` subcommand (assuming you are inside quickwit install directory):

> Let's make use of the default configuration file provided in quickwit installation throughout this guide by setting the `QW_CONFIG` environment variable.

Reviewer: ```suggestion
Let's make use of the default configuration file provided in Quickwit installation throughout this guide by setting the `QW_CONFIG` environment variable.
```

It should return 6 hits faster as Quickwit will query fewer splits.

## Distributed search

> Now that we have indexed our dataset and can do a local search, let's show how easy it is to start a distibuted search cluster with Quickwit. We will be launching a three nodes cluster.

Reviewer: ```suggestion
Now that we have indexed our dataset and can do a local search, let's show how easy it is to start a distributed search cluster with Quickwit. We will be launching a three-node cluster.
```

> First let's download the ssearche node configuration files:

Reviewer: ```suggestion
First, let's download the Searcher's configuration files:
```

```bash
for i in {1..3}; do 
curl -o searcher-$i.yaml https://raw.githubusercontent.com/quickwit-inc/quickwit/main/config/tutorials/hdfs-logs/searcher-$i.yaml
done
```

> Once the configuartion files are downloade, start a searcher node for each of the respective configuration file in a different terminal window.

Reviewer: ```suggestion
Once the configuration files are downloaded, start a searcher node for each of the respective configuration files in a different terminal window.
```

```bash
# run this in the first terminal window.
./quickwit service run searcher --config ./searcher-1.yaml
```

```bash
# run this in the second terminal window.
./quickwit service run searcher --config ./searcher-2.yaml
```

```bash
# run this in the third terminal window.
./quickwit service run searcher --config ./searcher-3.yaml
```

You will see in your terminal the confirmation that the instance has created or joined a cluster. Example of such a log:

```
INFO quickwit_cluster::cluster: Create new cluster. node_id="searcher-1" listen_addr=127.0.0.1:7280
```

:::note

Quickwit will use the configured `rest_listen_port` for serving the HTTP rest API via TCP as well as maintaining the cluster formation via UDP. Also, it will `{rest_listen_port} + 1` for gRPC communication between instances.

:::

Let's execute a simple query that returns only `ERROR` entries on field `severity_text` on one of our searcher node:

```bash
curl -v 'http://127.0.0.1:7280/api/v1/hdfs_logs/search?query=severity_text:ERROR
```

> Now that you have a search cluster running, ideally, you will want to load balance external requests to it. This can quickly be done using appropriate tool. with a load balancer added, you can now play with your cluster, kill processes randomly, add/remove new instances.

Reviewer: I don't think we need to add that as this is just a local cluster. This makes more sense on AWS.
